src/main/cpp/_nix_based/jssc.cpp

```diff
@@ -49,6 +50,13 @@
 
 //#include <iostream> //-lCstd use for Solaris linker
 
+// TODO: IMHO logging to slf4j would be nice.
+#ifndef DEBUG
+    #define DEBUG 0
+#endif
+#define LOG_WARN( fmt, ... ) fprintf(stderr, "[WRN %s:%d] " fmt, strrchr(__FILE__,'/')+1, __LINE__,  __VA_ARGS__ )
+#define LOG_DEBUG( fmt, ... ) if(DEBUG)fprintf(stderr, "[DBG %s:%d] " fmt, strrchr(__FILE__,'/')+1, __LINE__, __VA_ARGS__ );
```

Comment:
Is it normal to write to `stderr` for debug?

Reply:
I think this is the same question as [stackoverflow.com/q/4919093/4415884](https://stackoverflow.com/q/4919093/4415884).

**PRO**: Logging to stdout would make it hard (if not impossible) to write a program which produces data to stdout (Eg somewhat like `git archive > my.tar`) or also with pipes (Eg `tar cz * | ssh example.com -- tar xz`). If any library writes to stdout this will no longer be possible as it would "destroy" the actual data.

**CON**: AFAIK stderr is unbuffered. I never had an issue with this. But that does not mean it could have an impact which I'm not aware of.

**Finally**: In "my project" it does not matter as we do not use stdout for data. So "my project" would be fine with logging to stdout too. But I don't know about others. That's why I added the idea with slf4j. So the user could decide himself.

WDYT?